src/AllDayEvents/component.js

```diff
@@ -0,0 +1,224 @@
+      // is events for specific day in range
+      const dates = calculateDaysArray(initialDate, numberOfDays, rightToLeft);
+      const totalEvents = dates.map((date) => {
+        const dateStr = date.format(DATE_STR_FORMAT);
+        return eventsByDate[dateStr] || [];
+      });
+      const totalEventsWithPosition = this.getEventsWithPosition(totalEvents);
+      return totalEventsWithPosition;
+    },
+  );
+
+  onGridClick = (event, dayIndex) => {
+    const { initialDate, onGridClick, numberOfDays } = this.props;
+    if (!onGridClick) {
+      return;
+    }
+    const { locationY } = event.nativeEvent;
+    const hour = Math.floor(this.yToHour(locationY - CONTENT_OFFSET));
+
+    const adjustment = numberOfDays === 7 ? dayIndex - moment(initialDate).weekday() : dayIndex;
+
+    const date = moment(initialDate).add(adjustment, 'day').toDate();
+
+    onGridClick(event, hour, date);
+  };
+
+  render() {
+    const {
+      eventsByDate,
+      initialDate,
+      numberOfDays,
+      onEventPress,
+      onEventLongPress,
+      sendCallback,
+      eventContainerStyle,
+      EventComponent,
+      rightToLeft,
+    } = this.props;
+    const totalEvents = this.processEvents(
+      eventsByDate,
+      initialDate,
+      numberOfDays,
+      rightToLeft,
+    );
+
+    return (
+      <View style={styles.container}>
+        <View style={styles.events}>
+          {totalEvents.map((eventsInSection, dayIndex) => (
+            <TouchableWithoutFeedback
+              onPress={(e) => this.onGridClick(e, dayIndex)}
+              key={dayIndex}
+            >
+              <View style={[styles.event, { overflow: 'visible' }]}>
+                {eventsInSection.map((item) => (
+                  <View style={{ height: 45 }}>
+                    <Event
+                      key={item.data.id}
+                      event={item.data}
+                      position={item.style}
+                      onPress={onEventPress}
+                      onLongPress={onEventLongPress}
+                      EventComponent={EventComponent}
+                      containerStyle={eventContainerStyle}
+                      sendCallback={sendCallback}
+                    />
+                  </View>
+                ))}
+              </View>
+            </TouchableWithoutFeedback>
+          ))}
+        </View>
+      </View>
+    );
+  }
+}
+
+AllDayEvents.propTypes = {
+  numberOfDays: PropTypes.oneOf(availableNumberOfDays).isRequired,
+  eventsByDate: PropTypes.objectOf(PropTypes.arrayOf(Event.propTypes.event))
+    .isRequired,
+  initialDate: PropTypes.string.isRequired,
+  hoursInDisplay: PropTypes.number.isRequired,
+  onEventPress: PropTypes.func,
+  onEventLongPress: PropTypes.func,
+  sendCallback: PropTypes.func,
+  onGridClick: PropTypes.func,
+  eventContainerStyle: PropTypes.object,
+  EventComponent: PropTypes.elementType,
+  rightToLeft: PropTypes.bool,
+};
+
+export default AllDayEvents;
```

Comment:
Is this code newly written or adapted from somewhere else?

Reply:
This is essentially a copy of the `Event` component. The plan for this is to display "all day events" at the top of the component. It will require some decent modification, but I think this is a good starting point.